nats.go

```diff
@@ -1705,8 +1734,9 @@ func (nc *Conn) waitForMsgs(s *Subscription) {
 				}
 				continue
 			}
-			s.pMsgs--
-			s.pBytes -= len(m.Data)
+			msgLen = len(m.Data)
+			//			s.pMsgs--
+			//			s.pBytes -= len(m.Data)
 		}
 		mcb := s.mcb
 		max = s.max
```

Comment:
I believe there is a condition where with auto-unsub the code below may remove the sub but not update pMsgs which would cause Drain() to timeout since the value would not go to 0. Maybe in this case we need to artificially set it to 0 in code below that looks for a barrier message.

Reply:
Actually removeSub() will close the sub, and the drain checks on sub's closed boolean too so that is ok.